src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -934,6 +934,58 @@ public void ExceptionWhenTryingToBindToByteArray()
                 exception.Message);
         }
 
+        [Fact]
+        public void CanBindSingleElementToCollection()
+        {
+            var dic = new Dictionary<string, string>
+            {
+                {"MyString", "hello world"},
+                {"Nested:Integer", "11"},
+            };
+
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dic);
+            IConfiguration config = configurationBuilder.Build();
+
+            var stringArr = config.GetSection("MyString").Get<string[]>();
+            Assert.Equal("hello world", stringArr[0]);
+            Assert.Equal(1, stringArr.Length);
+
+            var stringAsStr = config.GetSection("MyString").Get<string>();
+            Assert.Equal("hello world", stringAsStr);
+
+            var nested = config.GetSection("Nested").Get<NestedOptions>();
+            Assert.Equal(11, nested.Integer);
+
+            var nestedAsArray = config.GetSection("Nested").Get<NestedOptions[]>();
+            Assert.Equal(11, nestedAsArray[0].Integer);
+            Assert.Equal(1, nestedAsArray.Length);
+        }
+
+        [Fact]
+        public void CannotBindSingleElementToCollectionWhenSwitchSet()
+        {
+            AppContext.SetSwitch("Microsoft.Extensions.Configuration.BindSingleElementsToArray", false);
```

Comment:
In a perfect world, this should use `RemoteExecutor`. Since this switch is process-wide, setting this switch can affect other tests running in the process.

Reply:
Did we really add a global switch for configuration behavior. This makes me sad 😢

Reply:
We discussed it at length on in API Review. The decision was that we didn't think people need/want the old behavior. Thus we didn't want to add a public API for it. But since this is a behavior change, we wanted a way to turn the behavior off, just in case someone needed it.

See https://github.com/dotnet/runtime/issues/57325#issuecomment-900508614